src/renderer/gdi/state.cpp

```diff
@@ -142,15 +142,6 @@ GdiEngine::~GdiEngine()
     _hwndTargetWindow = hwnd;
     _hdcMemoryContext = hdcNewMemoryContext;
 
-    // If we have a font, apply it to the context.
-    if (nullptr != _hfont)
-    {
-        LOG_HR_IF_NULL(E_FAIL, SelectFont(_hdcMemoryContext, _hfont));

```

Comment:
this code is not important?

Reply:
It works without this and allowed me to centralize the `FontHasWesternScript` call.
AtlasEngine doesn't set up any font related state until `UpdateFont` is called and it works properly. As such I'm sure that `SetHwnd` doesn't need to do that either. `UpdateFont` remains unmodified in this PR (apart from `_lastFontType` which is now part of `BeginPaint`).

My original intention was to write some code that ensures that `_fontHasWesternScript` is updated whenever `_lastFontType` is changed. Unfortunately GdiEngine is rather stateful and I was unable to come up with something that works properly. I'll try and see if I can do something about that.